### PR TITLE
add Collection>>atRandom:weighting: and cie and tests

### PR DESCRIPTION
**Pull request by privat**
This is a variation of `Collection>>atRandom` with a non-equiprobable way to choose elements.

In some language, the weights are given with a dictionary or with a sequence where the weights are in the same order.
Here I choose to ask a block. It feels more Pharo and more general.

### CODE REVIEW

File: src/Random-Core/Collection.extension.st [hunk 1]
```diff
@@ -29,3 +29,33 @@ Collection >> atRandom: aGenerator [
 	rand := aGenerator nextInteger: self size.
 	self withIndexDo: [:each :index | index = rand ifTrue: [^each]]
 ]
+
+{ #category : #'*Random-Core' }
+Collection >> atRandom: aRandom weighting: aBlock [
+
+	"Retrieve an element with a probability related to the weigting block.
```
Comment (Ducasse) [inline src/Random-Core/Collection.extension.st]: weighting

File: src/Random-Core/Collection.extension.st [hunk 2]
```diff
@@ -29,3 +29,33 @@ Collection >> atRandom: aGenerator [
...
+	The block takes and element as argument and is expected to provide a number as result.
```
Comment (Ducasse) [inline src/Random-Core/Collection.extension.st]: takes an element

File: src/Random-Core/Collection.extension.st [hunk 3]
```diff
@@ -29,3 +29,33 @@ Collection >> atRandom: aGenerator [
...
+	An implicit normalization is performed.
+	
+	Multiple occurences of equals elements multiply the probability to be choosen.
```
Comment (Ducasse) [inline src/Random-Core/Collection.extension.st]: I think that this is occurrences of equal elements

File: src/Random-Tests/OrderedDictionaryTest.extension.st [hunk 4]
```diff
@@ -8,3 +8,45 @@ OrderedDictionaryTest >> testAtRandom [
...
 ]
+
+{ #category : #'*Random-Tests' }
+OrderedDictionaryTest >> testAtRandomWeighting [
+
+	| random collection hits |
+	collection := { 0. 0. 1. 0. 4. 0. 3. 0. 2. 0. 0. }.
+
+	"Fixed randomness"
+	random := (Random seed: 42).
```
Comment (Ducasse) [inline src/Random-Tests/OrderedDictionaryTest.extension.st]: unnecessary parentheses

**Comment by privat** [comment]
fixed spelling and added more tests

**Review by MarcusDenker — APPROVED** [review]

**Pull request merged**